rewrite-maven/src/main/java/org/openrewrite/maven/cache/RocksdbMavenPomCache.java

```diff
@@ -103,22 +104,23 @@ static synchronized RocksCache getCache(String workspace) {
     CacheResult<MavenMetadata> UNAVAILABLE_METADATA = new CacheResult<>(CacheResult.State.Unavailable, null);
     CacheResult<MavenRepository> UNAVAILABLE_REPOSITORY = new CacheResult<>(CacheResult.State.Unavailable, null);
 
-    public RocksdbMavenPomCache(@Nullable File workspace) {
+    public RocksdbMavenPomCache(@Nullable String folder) {
```

Comment:
Note:
I'm not sure how to safely handle this change.

The AbstractRewriteMOJO uses the RocksdbMavenPomCache with:
`mavenParserBuilder.cache(new RocksdbMavenPomCache(Paths.get(System.getProperty("user.home"), ".rewrite", "cache", "pom").toFile()));`
and
`mavenParserBuilder.cache(new RocksdbMavenPomCache(Paths.get(pomCacheDirectory).toFile()))`

The former is pretty simple to fix, but later will cause `pomCacheDirectory` to be appended to 'System.getProperty("user.home") + "/.rewrite/cache".

Reply:
After these changes to code MOJO code would look like this:
```
                if (pomCacheDirectory == null) {
                    //Default directory is "~/.rewrite/cache/pom"
                    mavenParserBuilder.cache(new RocksdbMavenPomCache("pom"));
                } else {

                    mavenParserBuilder.cache(new RocksdbMavenPomCache(pomCacheDirectory));
                }
```